README.md

```diff
@@ -2,28 +2,88 @@
 
 Tools for working with Elsevier's Pure API.
 
-## Usage
+## Quick Start
 
-Please see `tests/test_*.py` for now. We freely admit this section of the documentation requires
-much improvement.
+If you set the `PURE_API_DOMAIN` and `PURE_API_KEY` environment variables,
+and want to use the latest Pure API version, using this package's client
+can be as easy as:
+
+```python
+from pureapi import client
+response = client.get('research-outputs')
+```
+
+For more control, pass a `client.Config` object. For example, to make requests
+to two different Pure API servers:
+
+```python
+from pureapi.client import Config
+response = client.get(
+    'persons',
+    config=Config(domain='example.com', key='123-abc')
+)
+test_repsonse = client.get(
+    'persons',
+    config=Config(domain='test.example.com', key='456-def', version='516')
+)
+```
+
+All functions that make requests of a Pure API server accept a `config`
+parameter. See the documentation for `client.Config` for more details.
+
+### Multi-Request Functions
+
+Many collections may contain too many records to download in a single request.
+For downloading large numbers of records in multiple requests, use the
+`client.*_all*()` functions. For example, to download all research outputs:
+
+```python
+for response in client.get_all('research-outputs'):
+   json = response.json()
+```
+
+### Record-Transforming Functions
+
+For even more convenience, each `client.*_all()` function has an associated
+`*_all_transformed()` function. Instead of yielding HTTP responses, which
+typicall contain many records each,these functions yield individual records,
+transformed from raw JSON into `addict.Dict` objects. These transformed records
+allow easier access to deeply nested fields, and may also ensure that some
+critical fields exist in each record, even if they may be `None` or empty by
+default.
+
+An example using one of these functions to get the titles of all research
+outputs:
+
+```python
+for ro in client.get_all_transformed('research-outputs'):
+   title = ro.title.value
+```
+
+For more details, see the documentation for each module. For more examples, see
+`tests/test_*.py`.
 
 ## Pure API Versions
 
-Successfully tested against Pure API versions 5.10.x - 5.12.x.
+Successfully tested against Pure API versions 5.16.x - 5.17.x.
 
 ## Requirements and Recommendations
 
 ### Python Versions
 
 pureapi requires Python >= 3.
 
-### Environment Variables
+### Pure API Domain and Key
 
-To connect to the Pure server, including when running `tests/test_client.py`, the
-`PURE_API_URL` and `PURE_API_KEY` environment variables must be set. One option is to set them in a 
-`.env` file. See `env.dist` for an example.
+The only configuration values the user absolutely must provide are a Pure API
```

Comment:
Because I didn't start from `env.dist`, I also needed to add `PURE_API_VERSION=517`. That might be missing from the readme